### ISSUE TITLE
session.test + symfony forms incompatibility 

### ISSUE THREAD

**Issue by theycallmeswift** [silexphp/Silex#872]
I'm writing some integration tests and I seem to have found an incompatibility between the `session.test` setup and Symfony forms (with csrf enabled).  

I have a route (`/users/new`) that renders a form:

```php
$form = $app['form.factory']->createBuilder('form', array())
  ->add('first_name')
  ->add('last_name')
  ->add('Sign Up', 'submit')
  ->getForm();

return $app['twig']->render('@UserBundle/users/new.html.twig', array(
  'form' => $form->createView(),
));
```

In my tests, I click a link that would render that form:

```php
$link = $crawler->filter('.login_area a[href="/users/new"]')->link();
$crawler = $client->click($link);
```

And that produces the following error:

```
1) \UserBundle\Tests\Features\UserLoginTest::testVisitorViewsSignUp
session_start(): Cannot send session cookie - headers already sent by (output started at /vendor/symfony/http-foundation/Symfony/Component/HttpFoundation/Response.php:298)

/vendor/symfony/form/Symfony/Component/Form/Extension/Csrf/CsrfProvider/DefaultCsrfProvider.php:70
/vendor/symfony/form/Symfony/Component/Form/Extension/Csrf/CsrfProvider/DefaultCsrfProvider.php:48
/vendor/symfony/form/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php:99
/vendor/symfony/form/Symfony/Component/Form/ResolvedFormType.php:214
/vendor/symfony/form/Symfony/Component/Form/ResolvedFormType.php:138
/vendor/symfony/form/Symfony/Component/Form/Form.php:974
/src/UserBundle/Controller/UsersController.php:51
/vendor/symfony/http-kernel/Symfony/Component/HttpKernel/HttpKernel.php:117
/vendor/symfony/http-kernel/Symfony/Component/HttpKernel/HttpKernel.php:61
/vendor/silex/silex/src/Silex/Application.php:516
/vendor/symfony/http-kernel/Symfony/Component/HttpKernel/Client.php:82
/vendor/symfony/browser-kit/Symfony/Component/BrowserKit/Client.php:324
/vendor/symfony/browser-kit/Symfony/Component/BrowserKit/Client.php:261
/src/UserBundle/Tests/Features/UserLoginTest.php:74
```

I traced it back to the following function in Symfony Form: https://github.com/symfony/Form/blob/master/Extension/Csrf/CsrfProvider/DefaultCsrfProvider.php#L71-82

```php
protected function getSessionId()
{
    if (version_compare(PHP_VERSION, '5.4', '>=')) {
        if (PHP_SESSION_NONE === session_status()) {
            session_start();
        }
    } elseif (!session_id()) {
        session_start();
    }

    return session_id();
}
```

Basically, Symfony is trying to enable the session for CSRF, but the session is a fake one.  We need to somehow set `session_status` to not be none.  Looks like this is specific to 5.4.
